allennlp/models/encoder_decoders/wikitables/wikitables_semantic_parser.py

```diff
@@ -360,23 +436,32 @@ def forward(self,  # type: ignore
                 # infinite action loop).
                 if i in best_final_states:
                     best_action_indices = best_final_states[i][0].action_history[0]
-                    credit = 0
+                    sequence_in_targets = 0
                     if target_action_sequences is not None:
                         # Use a Tensor, not a Variable, to avoid a memory leak.
                         targets = target_action_sequences[i].data
-                        credit = self._action_history_match(best_action_indices, targets)
-                    self._action_sequence_accuracy(credit)
+                        sequence_in_targets = self._action_history_match(best_action_indices, targets)
+
+                    self._action_sequence_accuracy(sequence_in_targets)
                     action_strings = [action_mapping[(i, action_index)] for action_index in best_action_indices]
                     try:
+                        self._has_logical_form(1.0)
                         logical_form = world[i].get_logical_form(action_strings, add_var_function=False)
                     except ParsingError:
```

Comment:
When would exception be raised? It looks like the `if` block this code is in executes only on finished states (or else `i in best_final_states` is `False`), which means `action_strings` should be a complete sequence, right?

Reply:
The code that mapped action sequences back to logical forms used to be a lot more buggy and would fail sometimes.  This doesn't really happen anymore, but I didn't remove the try/catch block, just in case.